scapy/layers/inet.py

```diff
@@ -216,6 +216,23 @@ class IPOption_Traceroute(IPOption):
                    IPField("originator_ip", "0.0.0.0")]
 
 
+class IPOption_Timestamp(IPOption):
+    name = "IP Option Timestamp"
+    optclass = 2
+    option = 4
+    fields_desc = [_IPOption_HDR,
+                   ByteField("length", 12),
```

Comment:
Thoughts: this field won't be updated if the conditional field is triggered...

Reply:
Yes it is true. "length" field won't be updated in case the "flg" field is 0. I didn't know how to it when I write IPOption_Timestamp class.
I found that I could use  post_build method to automatically calculate "length" field as in the following example:

    def post_build(self, p, pay):
        if self.length is None:
            l = len(p)
            p = p[:1] + struct.pack("!B", l) + p[2:]
        return p + pay

1. Is this the right method that I should use to calculate "length" field? If not, which is the right method?
2. Should I do a new pull request to include this functionality?

Reply:
I don't understand which is the problem why  a check failed: https://codecov.io/gh/secdev/scapy/compare/421eafcdeeb1c784e70c023c30bc27238ab936c0...e775ca3973d902474ba6c14a613cedcfe8e16a0d

How could I resolve it?

Reply:
The method you provided looks correct, it would be great to have in this PR so that we can merge it.
You can ignore the failing check

Reply:
@gpotter2 see my comment above.